server/channels/store/sqlstore/user_store.go

```diff
@@ -243,18 +243,23 @@ func (us SqlUserStore) Update(rctx request.CTX, user *model.User, trustedUpdateD
 	if !trustedUpdateData {
 		user.Roles = oldUser.Roles
 		user.DeleteAt = oldUser.DeleteAt
-	}
 
-	if user.IsOAuthUser() {
-		if !trustedUpdateData {
+		if user.IsOAuthUser() {
 			user.Email = oldUser.Email
 		}
-	} else if user.IsLDAPUser() && !trustedUpdateData {
-		if user.Username != oldUser.Username || user.Email != oldUser.Email {
-			return nil, store.NewErrInvalidInput("User", "id", user.Id)
+
+		if user.IsLDAPUser() {
+			if user.Username != oldUser.Username {
+				return nil, store.NewErrInvalidInput("User", "id", user.Id)
+			}
+			if user.Email != oldUser.Email {
+				return nil, store.NewErrInvalidInput("User", "email", user.Id)
+			}
+		}
+
+		if user.Email != oldUser.Email {
+			user.EmailVerified = false
 		}
-	} else if user.Email != oldUser.Email {
-		user.EmailVerified = false
 	}
```

Comment:
One problem with the implementation is that existing users who have an unverified email address don't get a fix.

We could add the following code here:
```suggestion
	}
	if user.IsLDAPUser() || user.IsOAuthUser() {
		user.EmailVerified = true
	}
```

Would that be too much business logic on the store layer? Alternatively, we could do a full app-layers migration, but that would be a lot of work.

Would love to hear your thoughts.

Reply:
If that's a hard rule, I think you could add it here.

Reply:
As we are already performing actions(buisness logic) based on the user's AuthService, it should be alright to do add this check as well.
Furthermore, I think we can just use the `IsSSOUser` method, instead of separate checks. Also, just to keep it consistent and avoid confusion, it would be good to add the same check in the Store `Save` method as well.